libafl/src/bolts/shmem.rs

```diff
@@ -979,3 +979,88 @@ pub mod win32_shmem {
         }
     }
 }
+
+/// A cursor around [`ShMem`] that immitates [`std::io::Cursor`]. Notably, this implements [`Write`] for [`ShMem`] in std environments.
+pub struct ShMemCursor<T: ShMem> {
+    inner: T,
+    pos: usize,
+}
+
+#[cfg(feature = "std")]
+impl<T: ShMem> ShMemCursor<T> {
+    pub fn from_shmem(shmem: T) -> Self {
```

Comment:
I guess this could `impl From for ShMemCursor`, right? Not sure if it would be better though

Reply:
true. I'm not sure what's the "official" style guide here. when should from be implemented?

for reference: std::io::cursor [implements just a single](https://doc.rust-lang.org/std/io/struct.Cursor.html#method.new) "new" method with a single parameter and no from/into at all. So I suppose I should rename the method to simply "new"?

Reply:
Makes sense to mimick that behavior, then(?)

Reply:
yes :) I also changed it in a few other places. seems like I have tendency to call these methods `from_*` instead of `new`...

Reply:
If you want a `from`, just implement the `From` trait, then rust will do the conversation for you with `blubber.into()`